samza-core/src/main/scala/org/apache/samza/storage/ContainerStorageManager.java

```diff
@@ -729,11 +1058,16 @@ public void stop() {
      * can invoke compaction.
      */
     public void stopPersistentStores() {
-      this.taskStores.values().stream().filter(storageEngine -> {
-          return storageEngine.getStoreProperties().isPersistedToDisk();
-        }).forEach(storageEngine -> {
-            storageEngine.stop();
-          });
+
+      Map<String, StorageEngine> persistentStores = this.taskStores.entrySet().stream().filter(e -> {
```

Comment:
Will this stop side input stores? It is possible for side input stores to be persisted to disk.

Reply:
No this wont, note that this is inside TaskRestoreManager, which only handles changelogged stores.